packages/components/src/test/properties/UniquePropertyValuesSelector.test.tsx

```diff
@@ -347,6 +347,45 @@ describe("UniquePropertyValuesSelector", () => {
       expect(actualClassName).to.be.equal(expectedClassName);
     });
 
+    it("calls 'getPagedDistinctValues' with ruleset that is created from a 'NestedContentField' with multiple layers of nesting", async () => {
+      const testProperty = {
+        name: "#testField",
+        displayLabel: "propertiesField",
+        typename: "number",
+      };
+
+      const relationshipPath = createTestRelationshipPath();
+      const lastStepOfRelationshipPath: RelatedClassInfo = createTestRelatedClassInfo({
+        targetClassInfo: createTestECClassInfo({ name: "testSchema:testClass" }),
+      });
+      relationshipPath.push(lastStepOfRelationshipPath);
+
+      // create the field that is checked and set its 'grandparent' to contain the pathToPrimaryClass
+      const testField = createTestNestedContentField({ name: "testField", nestedFields: [] });
```

Comment:
`UniquePropertyValuesSelector` is working with `PropertiesField`s. I think these new tests could improved by creating field hierarchy that looks like this:
```
grandParentField [NestedContentField]
| - parentFiled [NestedContentField]
    | - propertyField [PropertiesField]
```

Also `testDescriptor` should contain the root field e.g. `grandParentField` in order to match real usecase.